### DISCUSSION TITLE
`allowedPostUpgradeCommands` is not respected or interpreted correctly

### DISCUSSION THREAD

**Discussion by moltar** [renovatebot/renovate#17906] [Request Help] [answered]
### How are you running Renovate?

Mend Renovate hosted app on github.com

### Was this something which used to work for you, and then stopped?

I never saw this working

### Describe the bug

I have the following config:

```json
{
  "$schema": "https://docs.renovatebot.com/renovate-schema.json",
  "extends": [
    "@our-org-scope"
  ],
  "allowedPostUpgradeCommands": ["^npx patch-package "],
  "packageRules": [
    {
      "matchPackageNames": [
        "aws-amplify"
      ],
      "postUpgradeTasks": {
        "commands": [
          "npx patch-package @aws-amplify/auth"
        ],
        "fileFilters": [
          "patches/*.patch"
        ]
      }
    }
  ]
}
```

According to the [docs](https://docs.renovatebot.com/self-hosted-configuration/#allowpostupgradecommandtemplating), `allowedPostUpgradeCommands` option expects a regex, which I have provided:

```
^npx patch-package 
```

But when renovate runs, it reports an error on the PR:

> Post-upgrade command 'npx patch-package @aws-amplify/auth' has not been added to the allowed list in allowedPostUpgradeCommands

Any ideas?

Thanks!

### Relevant debug logs

<details><summary>Logs</summary>

```
Post-upgrade command 'npx patch-package @aws-amplify/auth' has not been added to the allowed list in allowedPostUpgradeCommands
```

</details>

### Have you created a minimal reproduction repository?

No reproduction repository

**Comment by viceice** [comment, marked as answer]
`allowedPostUpgradeCommands` can't be used in repo config. needs to be in global bot config. All options from self-hosting section can only be used in global config.

https://docs.renovatebot.com/self-hosted-configuration/

**Reply by moltar** [reply]
Thanks for the super quick response, @viceice!

Does that mean that post upgrade commands are not available in the hosted version at all?

**Reply by viceice** [reply]
yes

**Reply by moltar** [reply]
Thank you!
